full-backend-tests/src/test/java/org/graylog/plugins/views/ScriptingApiResourceIT.java

```diff
@@ -119,6 +121,36 @@ public void beforeAll() {
         api.fieldTypes().waitForFieldTypeDefinitions( "source", "facility", "level");
     }
 
+    @ContainerMatrixTest
+    void testAggregationByStream() {
+        final ValidatableResponse validatableResponse = given()
+                .spec(requestSpec)
+                .when()
+                .body("""
+                         {
+                           "group_by": [
+                             {
+                               "field": "streams"
+                             }
+                           ],
+                           "metrics": [
+                             {
+                               "function": "count"
+                             }
+                           ]
+                        }
+                         """)
+                .post("/search/aggregate")
+                .then()
+                .log().ifStatusCodeMatches(not(200))
+                .statusCode(200);
+
+        validatableResponse.assertThat().body("datarows", Matchers.hasSize(3));
```

Comment:
I'm not sure if you shouldn't verify the whole response with one assertion.
If first assertion fails, you will know smth is wrong, but you will not have the full picture, you will just learn that you miss one stream. But if you believe it is enough, we can leave it as it is.

Reply:
I like to add a `.log().ifValidationFails()` for those cases, so I will get the response body when an assertion fails.

Reply:
Thanks, added logging Dennis suggested :+1: